### PR TITLE
Feature/13 Registratration new user  based Element UI

### PR DESCRIPTION
**Pull request by IgorNazaryok**
the following windows are implemented
Registration 2
Registration
Registration success

mailbox is busy

### CODE REVIEW

**Comment by horba** [comment]
1. маржины и паддинги не соответсвуют дизайну. неоправдоно большие расстояние между элэментами формы. 
2. валдационное сообщение слишком большое. В идеале все-таки сделать еррор как на дизайне. или хотя бы приблизить. хотя бы просто небольшим шрифтом в спане красного цвета текст ошибки.
3. надпись. Уже есть аккаунт? Вход - не по дизайн - жирным только слово вход
4. Кнопка не того цвета. по дизайну - `#6663FF` (собственно, наш праймари цвет), по скринам `#8582ff`

**Comment by horba** [comment]
по скринам не вижу где находится кнопка. она находится в меню? у меня по коду есть подозрения, что нет

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.close {
+  position: absolute;
+  left: 88.50%;
+  top: 11px;
+}
+.btn_padding {
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css]: kebab-case
read our wiki https://github.com/horba/film-finity/wiki/Frontend-Namings-and-Structure

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  align-items: flex-end;
+  text-align: center;
+  color: #292c33;
+}
+.entrance {
+  font-weight: 600;
+  padding: 30px;
+  font-size: 18px;
+}
+.accountCreate{
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css]: kebab-case
read our wiki https://github.com/horba/film-finity/wiki/Frontend-Namings-and-Structure

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+.item {
+height: 95px;
+}
+.text {
+  font-weight: 600;
+  font-size: 24px;
+  line-height: 144%;
+  align-items: flex-end;
+  text-align: center;
+  color: #292c33;
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css]: use css variables

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+.btn_padding {
+  letter-spacing: 1px;
+}
+.item {
+height: 95px;
+}
+.text {
+  font-weight: 600;
+  font-size: 24px;
+  line-height: 144%;
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css]: no percents

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 5]
```diff
@@ -0,0 +1,30 @@
+.close {
+  position: absolute;
+  left: 88.50%;
+  top: 11px;
+}
+.btn_padding {
+  letter-spacing: 1px;
+}
+.item {
+height: 95px;
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css]: tab

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 6]
```diff
@@ -0,0 +1,30 @@
+.close {
+  position: absolute;
+  left: 88.50%;
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css]: no percents

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 7]
```diff
@@ -0,0 +1,30 @@
+.close {
+  position: absolute;
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css:2]: are you sure that it still actual?

Comment (horba) [reply front/src/components/RegistrationUser/RegistrationUser.component.css:2]: element ui provide own close button, i sure

File: front/src/components/RegistrationUser/RegistrationUser.component.css [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  font-weight: 600;
+  padding: 30px;
+  font-size: 18px;
+}
+.accountCreate{
+  text-align: center;
+  margin: 15px;
+  line-height: 2.0;
+  height: 450px;
+}
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.css:40]: incorrect file name. please read guide https://github.com/horba/film-finity/wiki/Frontend-Namings-and-Structure
components/registration-dialog/registration-dialog.component.css
components/registration-dialog/registration-dialog.component.html
components/registration-dialog/registration-dialog.component.js
components/registration-dialog/registration-dialog.component.vue

File: front/src/components/RegistrationUser/RegistrationUser.component.html [hunk 9]
```diff
@@ -0,0 +1,73 @@
+  <div>   
+    <el-button type="primary" @click="VisibleRegistrationForm(true)">Зарегистрироваться</el-button>
+    <el-dialog 
+    title="Создать аккаунт"
+    :visible.sync="isVisibleRegistrationForm" 
+    center
+    :show-close="false"
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.html:7]: why do you skip close button in dialog in element ui and provide your own instead? i sure it makes extra code and styles

File: front/src/components/RegistrationUser/RegistrationUser.component.html [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        >
+        </el-input>
+        </el-form-item>
+      </el-form>
+      
+      <span slot="footer" class="dialog-footer">
+        <el-button
+            id='Registration'
+            type="primary"
+            @click="Registration()"
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.html:41]: register - not registration, read article about naming in wiki https://github.com/horba/film-finity/wiki/JS-Code-Style-Guide

register() - in lowerCase read our guide https://github.com/horba/film-finity/wiki/Frontend-Namings-and-Structure

File: front/src/components/RegistrationUser/RegistrationUser.component.html [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+            :disabled='isDisabled'
+          >Зарегистрироваться</el-button> 
+          <div class="entrance">
+            Уже есть аккаунт? Вход
+          </div>     
+      </span>
+    </el-dialog>
+    <el-dialog 
+      title="Создать аккаунт"
+      :visible.sync="isVisibleRegistrationSuccess"
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.html]: might, just isRegistrationSuccess or other one

File: front/src/components/RegistrationUser/RegistrationUser.component.html [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+      </span>
+    </el-dialog>
+    <el-dialog 
+      title="Создать аккаунт"
+      :visible.sync="isVisibleRegistrationSuccess" 
+      center
+      :show-close="false"
+      width= "464px"
+    >
+    <div class="close"><el-button icon="el-icon-close" circle @click="VisibleRegistrationSuccess(false)"></el-button> </div>
```
Comment (horba) [inline front/src/components/RegistrationUser/RegistrationUser.component.html]: it seems like  overcoding

File: front/src/store/modules/ReristrationUser.store.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+      return state.mailboxBusy;
+    },
+    getRegistrationSuccess (state) {
+      return state.isVisibleRegistrationSuccess;
+    },
+    getRegistrationForm (state) {
+      return state.isVisibleRegistrationForm;
+    }
+  }
+};
```
Comment (horba) [inline front/src/store/modules/ReristrationUser.store.js:21]: registration.store.js

File: front/src/store/index.js [hunk 14]
```diff
@@ -13,6 +14,6 @@ export default new Vuex.Store({
   actions: {
   },
   modules: {
-    сelebrities
+    сelebrities, ReristrationUser
```
Comment (horba) [inline front/src/store/index.js:17]: wasn't it cinfusing for you that celebrities starts from small letter and your from upper?
please, be more careful with naming. we have standard

File: server/Entities/Models/Users.cs [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebAPI.Models
+{
+    public class User
+    {
+        public int Id { get; set; }
+        public string UserName { get; set; }
```
Comment (horba) [inline server/Entities/Models/Users.cs:12]: just Name

File: server/Entities/Models/Users.cs [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+        public string Email { get; set; }
+        public string UserPassword { get; set; }
```
Comment (horba) [inline server/Entities/Models/Users.cs:14]: just Password

it is property of User entity - we don`t need to specify that it belongs user - it is obvious. we specify belonging if it diffs with entity owner

File: front/src/styles/common.css [hunk 17]
```diff
@@ -21,3 +21,6 @@ button.second-color {
 .main-content {
     min-height: calc(100vh - (var(--header-height) + var(--footer-height) + var(--footer-bottom-height)));
   }
+* {
+    font-family: Arial, Helvetica, sans-serif;
```
Comment (horba) [inline front/src/styles/common.css]: is it the font from design? why do you add it?

File: front/src/store/modules/ReristrationUser.store.js [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+        data: {
+          userName: newUser.userName,
+          email: newUser.email,
+          userPassword: newUser.userPassword
+        }
+      }).then(response => {
+        if (Number(response.data)) {
+          commit('updateMailboxBusy', !response.data);
+          commit('updateVisibleRegistrationForm', !response.data);
+          commit('updateVisibleRegistrationSuccess', response.data);
```
Comment (horba) [inline front/src/store/modules/ReristrationUser.store.js]: you see, it has problems. 
your store logic is connected with details of component. each change in design or naming in component or other things will cause the changes in store. it is high cohesion 
it is better to store just something like isRegistrationSuccess and based on this to make some other things in your components. Remember, store is more business logic layer. variable name should reflect its meaning not form

File: front/src/store/modules/ReristrationUser.store.js [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+      axios({
+        method: 'post',
+        url: 'https://localhost:44333/api/user',
+        data: {
+          userName: newUser.userName,
+          email: newUser.email,
+          userPassword: newUser.userPassword
+        }
+      }).then(response => {
+        if (Number(response.data)) {
```
Comment (horba) [inline front/src/store/modules/ReristrationUser.store.js]: it is really confusing
why we check that response is number.

See, response from server it is status code which means http status of success and unsuccessful with some types of error. and data. 

I sure you should change the model of response. If you want to return unsuccessful result it should be something like {isSuccess: false, isEmailExist: true} and handle it in component. Or other option, server return error and error status specific for unsuccessful registration you check it in store and return in promise {isSuccess: false, isEmailExist: true} and component handles it

File: front/src/store/modules/ReristrationUser.store.js [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+  getters: {
+    getMailboxBusy (state) {
+      return state.mailboxBusy;
+    },
+    getRegistrationSuccess (state) {
+      return state.isVisibleRegistrationSuccess;
+    },
+    getRegistrationForm (state) {
+      return state.isVisibleRegistrationForm;
+    }
```
Comment (horba) [inline front/src/store/modules/ReristrationUser.store.js]: don't use getters just to return state.
read docs of vuex. getters - for some ways to process in reactive way data in state

File: server/WebAPI/Controllers/UserController.cs [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+        }
+        [HttpPost]
+        public async Task<ActionResult<int>> Post(User value)
+        {
+            User user = await dbContext.Users.FirstOrDefaultAsync(x => x.Email == value.Email);
+            if (user == null)
+            {
+                dbContext.Users.Add(value);
+                await dbContext.SaveChangesAsync();
+                return 1;
```
Comment (horba) [inline server/WebAPI/Controllers/UserController.cs]: it is really bad way. Please read my comment for store
also, read articles about http results

and, it seems you didn't read the article in educational channel with example of code

**Review by horba — CHANGES_REQUESTED** [review]
- code styles problems
- extra code and styles
- wrong naming of variables and files
- percentages in styles
- wrong colors and not usage of variables

Also:
1. problems with architecture of store
2. problems with structure of server response
3. it could be good point - separate different steps of registration by different components, it is optional

It seems you didn't read the article in education, please read or read again

File: front/src/components/RegistrationUser/RegistrationUser.component.css (hunk 1 above)
Comment (IgorNazaryok) [reply front/src/components/RegistrationUser/RegistrationUser.component.css]: fixed

File: front/src/components/RegistrationUser/RegistrationUser.component.css (hunk 2 above)
Comment (IgorNazaryok) [reply front/src/components/RegistrationUser/RegistrationUser.component.css]: fixed

File: front/src/components/RegistrationUser/RegistrationUser.component.css (hunk 3 above)
Comment (IgorNazaryok) [reply front/src/components/RegistrationUser/RegistrationUser.component.css]: fixed

File: front/src/components/RegistrationUser/RegistrationUser.component.css (hunk 4 above)
Comment (IgorNazaryok) [reply front/src/components/RegistrationUser/RegistrationUser.component.css]: fixed

File: front/src/styles/common.css (hunk 17 above)
Comment (IgorNazaryok) [reply front/src/styles/common.css]: removed

File: front/src/store/modules/ReristrationUser.store.js (hunk 18 above)
Comment (IgorNazaryok) [reply front/src/store/modules/ReristrationUser.store.js]: I do not understand, please read more. Example?

**Pull request closed without merging**